lib/index.js

```diff
@@ -348,11 +348,20 @@ I18N.prototype.translate = function() {
 
 	cacheable = ! options.components;
 	if ( cacheable ) {
-		optionsString = JSON.stringify( options );
-		translation = this.state.translations.get( optionsString );
-		// Return the cached translation.
-		if ( translation ) {
-			return translation;
+		// Safe JSON stringification here to catch Circular JSON error
+		// caused by passing a React component into args where only scalars are allowed
+		try {
+			optionsString = JSON.stringify( options );
```

Comment:
this doesn't generate a deterministic cache key. the order of the keys depends on how the options object was created.

Reply:
> this doesn't generate a deterministic cache key. the order of the keys depends on how the options object was created.

Does the order of the properties matter here? Do you recommend we change anything before release a new version in #62?

Reply:
of course, if you don't want to pull a new dependency and the options object has a consistent shape, you could iterate over it by hand and do the same thing.